scripts/environments/Environment.py

```diff
@@ -41,17 +41,25 @@ def __init__(self, env_config: EnvironmentConfig, gripper_config: GripperConfig)
 
         self.observation_type = env_config.observation_type
         self.object_type = env_config.object_type
+        self.action_type = env_config.action_type
 
         self.goal_selection_method = env_config.goal_selection_method
         self.noise_tolerance = env_config.noise_tolerance
 
         self.aruco_detector = ArucoDetector(marker_size=env_config.marker_size)
-        self.object_marker_id = self.gripper.num_motors + 3  # Num Servos + Finger Tips (2) + 1
+        if self.gripper.num_motors == 9:
+            self.object_marker_id = 4 # hardcoded for 3 finger gripper for now
+        else: 
+            self.object_marker_id = self.gripper.num_motors + 3  # Num Servos + Finger Tips (2) + 1
 
-        self.goal_state = self.get_object_state()
+        if self.gripper.actuated_target is not None:
+            self.goal_state = self.gripper.current_object_position()#self.get_object_state()
+        else:
```

Comment:
the whole point of get_object_state() is to avoid these if statements being here - get_object_state() should be handle the potential combinations and return the required values generically - this shouldn't be split into an if statement